### ISSUE TITLE
add Shape selection event

### ISSUE THREAD

**Issue by OnionKiller** [napari/napari#6886]
## 🚀 Feature
selected event for shapes in Shape layer

## Motivation

When selecting a shape (Poligon, Line, etc.) it is uesful if there is an event which can capture the fact this happend.

## Pitch

I am trying to implement a segmentation visualization where the generated segments are displayed as Polygon shapes in a Layer.
I would like to create a visualization, which displays specific information of a single Polygon, selected by the user.

## Alternatives

Mouse click event, especially the doubleclick event can be captured, and by finding if the ursor position is inside a Shape, te event can be possibly deduced, but it is not ideal.

## Additional context

The request was mentioned [here](https://forum.image.sc/t/shapes-layer-selection-events/95708) originally.
I'll try to implement this feature, @brisvag could you please help where to start?

**Comment by melonora** [comment]
I don't actually think one could use the highlight as I also had the case that hovering over a shape highlights the shape. @OnionKiller would you be interested in doing paired coding to implement this?

**Comment by OnionKiller** [comment]
@melonora Yes, I would happily do the pair programming session. I'll contact you on the Napari Zulip server if thats okay for you.

**Comment by melonora** [comment]
Great!

**Comment by multimeric** [comment]
Was there any progress on this? I can attest that it would be a useful event. `highlight` is okay for now in that it does trigger when the selection changes, it just also triggers at other times. So as long as you have logic to handle that, it's okay.

There is a `selection` event as well as a `selected_properties` but neither worked for me. They didn't trigger in all cases I needed.

**Comment by multimeric** [comment]
If anyone is interested, I implemented a mini `EventEmitter` that fires whenever your shape selection changes. It's probably not as optimised as a built-in solution, but it works fine for my usage. Feel free to use (within terms of the license). You can just copy the file, which is self contained.

https://github.com/multimeric/napari_lattice/blob/523e2ba353366002ed02d4bb9137876238e3edd3/plugin/napari_lattice/shape_selection.py

**Comment by jo-mueller** [comment]
Is someone still working on this? Would be nice to have it - could also try myself at it.

**Comment by Czaki** [comment]
I do not know about anyone working on this, so feel free to open PR.

**Comment by jo-mueller** [comment]
Okay, I think I got it. PR opened 👍
